### DISCUSSION TITLE
good voice search with out LLM. (trying to make a mass proof of concept)

### DISCUSSION THREAD

**Discussion by Bram-diederik** [music-assistant/support#2417] [Feature requests and ideas]
Heya. I have scripted a system together that uses a Levenshtein query in sql to play the best found artist in mpd. 

mpd is quite limited because hass can only play playlists. so i have a playlist generated for every artist. SO no play album of artist. 

Could this be added to music assistant.  marcelveldt knows im trying to get a proof of concept to work for mass. but i have issues.

I cant get local music queried with the advanced mass search. (So dont get a list of local artist nor there albums for the db) 
to get the best result the list with artist should be limited. so only local music (with 3 or more tracks for example). (or most played artists on spotify)

I think this is a good feature to get working one way or the other for the environment and no internet possibilities. 

https://en.wikipedia.org/wiki/Levenshtein_distance
https://stackoverflow.com/questions/560709/levenshtein-distance-in-t-sql
https://community.home-assistant.io/t/play-mpd-playlists-podcasts-and-internet-radio-with-out-ai-v0-5/690855

**Comment by OzGav** [comment]
Limiting mass.search to the in library items is on the to do list

**Comment by OzGav** [comment]
Is the `get_library `action a solution to this?
